docs/integration-guide.md

```diff
@@ -107,6 +107,342 @@ Here's how we recommend doing it:
+
+Once you have written some tests, you can run just these new tests from within
+the Zulip dev environment with this command:
+
+```
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:/srv/zulip$
+./tools/test-backend zerver.tests.test_hooks.HelloWorldHookTests
+```
+
+(Note: You must run the tests from `/srv/zulip` directory.)
+
+You will see some script output and if all the tests have passed, you will see:
+
+```
+Running zerver.tests.test_hooks.HelloWorldHookTests.test_hello_message
+DONE!
+```
+
+#### Step 4: Create documentation
+
+Next, we add end-user documentation for our webhook integration to
+`templates/zerver/integrations.html`.
+
+First, add a `div` that displays the logo of your integration and a link to its
+documentation:
+
+```
+ <div class="integration-lozenge integration-helloworld">
+   <a class="integration-link integration-helloworld" href="#helloworld">
+      <img class="integration-logo" src="/static/images/integrations/logos/helloworld.png" alt="Hello World logo" />
+      <span class="integration-label">Hello World</span>
+   </a>
+ </div>
+```
+
+And second, a div with the usage instructions:
+
+```
+<div id="helloworld" class="integration-instructions">
+
+    <p>Learn how Zulip integrations work with this simple Hello World example!</p>
+
+    <p>The Hello World webhook will use the <code>test<code> stream, which is
+    created by default in the Zulip dev environment. If you are running
+    Zulip in production, you should make sure this stream exists.</p>
+
+    <p>Next, on your <a href="/#settings" target="_blank">Zulip
+    settings page</a>, create a Hello World bot.  Construct the URL for
+    the Hello World bot using the API key and stream name:
+      <code>{{ external_api_uri }}/v1/external/helloworld?api_key=abcdefgh&amp;stream=test</code>
+    </p>
+
+    <p><b>Congratulations! You're done!</b><br /> Your messages may look like:</p>
+
+    <img class="screenshot" src="/static/images/integrations/helloworld/001.png" />
+</div>
+```
+
+Both blocks should fall alphabetically so we add these two divs between the
+blocks for Github and Hubot, respectively.
+
+See [Documenting your integration](#documenting-your-integration) for further
+details, including how to easily create the message screenshot.
+
+#### Step 5: Preparing a pull request to zulip/zulip
+
+When you have finished your webook integration and are ready for it to be
+available in the Zulip product, follow these steps to prepare your pull
+request:
+
+1. Run tests including linters and ensure you have addressed any issues they
+   report. See [Testing](testing.html) for details.
+2. Read through [Code styles and conventions](code-style.html) and take a look
+   through your code to double-check that you've followed Zulip's guidelines.
+3. Take a look at your git history to ensure your commits have been clear and
+   logical (see [Version Control](version-control.html) for tips). If not,
+   consider revising them with `git rebase --interactive`.
```

Comment:
I'd probably mention that for most webhooks, this means squashing the changes into a single commit and giving it a good, clear commit message.